src/protostar_cli.py

```diff
@@ -0,0 +1,100 @@
+# pylint: disable=no-self-use
+import sys
+from logging import INFO, Logger, StreamHandler, getLogger
+from pathlib import Path
+from typing import Any
+
+from src.cli import CLIApp, Command
+from src.commands import (
+    BuildCommand,
+    InitCommand,
+    InstallCommand,
+    RemoveCommand,
+    TestCommand,
+    UpdateCommand,
+    UpgradeCommand,
+)
+from src.protostar_exception import ProtostarException
+from src.utils import (
+    Project,
+    ProtostarDirectory,
+    StandardLogFormatter,
+    VersionManager,
+    log_color_provider,
+)
+
+
+class ProtostarCLI(CLIApp):
+    @classmethod
+    def create(cls, script_root: Path):
+        protostar_directory = ProtostarDirectory(script_root)
+        version_manager = VersionManager(protostar_directory)
+        project = Project(version_manager)
+        return cls(script_root, protostar_directory, project, version_manager)
+
+    def __init__(
+        self,
+        script_root: Path,
+        protostar_directory: ProtostarDirectory,
+        project: Project,
+        version_manager: VersionManager,
+    ) -> None:
+        self.project = project
+        super().__init__(
+            commands=[
+                InitCommand(script_root, version_manager),
+                BuildCommand(project),
+                InstallCommand(project),
+                RemoveCommand(project),
+                UpdateCommand(project),
```

Comment:
We considered registration using a decorator, is that still on the table or is that impossible?

Reply:
In this case decorators are not the best choice because we need to pass dependencies (Project, VersionManager etc.).

Reply:
I dont understand why it makes using decorators a problem, you can register classes instead of objects

Reply:
You can do it in different PR though

Reply:
How am I supposed to provide dependencies to the registered classes without using some DI framework?

Reply:
Python is really dynamic so it won't be hard, not a concern of this PR though. Can you make task for this?

Reply:
It's overengineering at this point.
https://github.com/software-mansion/protostar/issues/245

Reply:
Agreed